esp-hal/src/i2c.rs

```diff
@@ -195,12 +209,12 @@ where
 {
     /// Reads enough bytes from slave with `address` to fill `buffer`
     pub fn read(&mut self, address: u8, buffer: &mut [u8]) -> Result<(), Error> {
-        self.peripheral.master_read(address, buffer)
+        maybe_recover!(self, { self.peripheral.master_read(address, buffer) })
```

Comment:
This is a very confusing way of writing code similar to

```rust
self.peripheral_master_read(address, buffer).or_else(|err| { self.recover(); err})
```

Can we make do without the macro?

Reply:
Matter of taste but sure - no preference

Reply:
To me the macro reads like "if self is an error, call the block" which is obvious nonsense 🤷‍♂️